src/bundle/Resources/public/js/scripts/udw/select.location.js

```diff
@@ -0,0 +1,94 @@
+                },
+            },
+        });
+
+        const request = new Request('/api/ezp/v2/views', {
+            method: 'POST',
+            headers: {
+                Accept: 'application/vnd.ez.api.View+json; version=1.1',
+                'Content-Type': 'application/vnd.ez.api.ViewInput+json; version=1.1',
+                'X-Requested-With' : 'XMLHttpRequest',
+                'X-Siteaccess': siteaccess,
+                'X-CSRF-Token': token
+            },
+            body,
+            mode: 'same-origin',
+            credentials: 'same-origin',
+        });
+
+        const errorMessage = Translator.trans(
+            /*@Desc("Cannot find children locations with given id list")*/ 'select_location.error',
+            {},
+            'universal_discovery_widget'
+        );
+        fetch(request)
+            .then(window.eZ.helpers.request.getJsonFromResponse)
+            .then((json) => callback(json))
+            .catch(() => window.eZ.helpers.notification.showErrorNotification(errorMessage));
+    };
+
+    const removeRootFromPathString = (pathString) => {
+        const pathArray = pathString.split('/').filter((val) => val);
+
+        return pathArray.splice(1, pathArray.length - 1);
+    };
+    const closeUDW = () => ReactDOM.unmountComponentAtNode(udwContainer);
+    const onConfirm = (btn, items) => {
+        closeUDW();
+
+        const pathString = items[0].pathString;
+        const displayWrapper = doc.querySelector(btn.dataset.displayWrapper);
+        const pathStringInput = doc.querySelector(btn.dataset.locationPathInputSelector);
+
+        pathStringInput.value = pathString;
+        pathStringInput.dispatchEvent(new Event('change'));
+
+        if (!displayWrapper) {
+            return;
+        }
+
+        findLocationsByIdList(removeRootFromPathString(pathString), (data) => {
+            const searchHitList = data.View.Result.searchHits.searchHit;
+            const pathStringNames = searchHitList.map((searchHit) => {
+                return searchHit.value.Location.ContentInfo.Content.Name;
+            }).join(' / ');
+
+            displayWrapper.querySelector(btn.dataset.selectedContentNameSelector).innerHTML = pathStringNames;
+        });
+    };
+    const onCancel = () => closeUDW();
+    const openUDW = (event) => {
+        event.preventDefault();
+
+        const config = JSON.parse(event.currentTarget.dataset.udwConfig);
+
+        ReactDOM.render(React.createElement(global.eZ.modules.UniversalDiscovery, Object.assign({
+            onConfirm: onConfirm.bind(null, event.currentTarget),
+            onCancel,
+            title: event.currentTarget.dataset.universalDiscoveryTitle,
+            multiple: false,
+            startingLocationId: window.eZ.adminUiConfig.universalDiscoveryWidget.startingLocationId,
+            restInfo: { token, siteaccess }
+        }, config)), udwContainer);
+    };
+
+    btns.forEach(btn => btn.addEventListener('click', openUDW, false));
```

Comment:
I think we are missing `clear selected subtree`? Because I don't see any JS for it.
<img width="386" alt="screenshot at nov 15 14-32-36" src="https://user-images.githubusercontent.com/12594013/48556145-608eed80-e8e3-11e8-8f23-a2433b3728ac.png">

Reply:
I left that out as redesign is not in scope of this PR. It will be done by @tischsoic

Reply:
I will provide PR with styling after merge of this one, because there will be also changes in other places.

Reply:
Why not push changes into this branch? We will have a better overview of changes and we will be able to perform proper QA on this.

Reply:
Because it involves restyling whole filters menu, not only Sub-tree filter.

Reply:
Do you mean that only styling is out of scope of this PR or also some core actions related to the new feature? In the latter case completing QA will not be possible here. We had a similar case here 
https://github.com/ezsystems/ezplatform-admin-ui/pull/697 and that's why it wasn't merged.

Reply:
In offline discussion with @ViniTou we decided that we I will push my changes to this PR. :)